bin/cml-runner.js

```diff
@@ -326,9 +351,15 @@ const opts = yargs
     'idle-timeout',
     'Time in seconds for the runner to be waiting for jobs before shutting down. Setting it to 0 disables automatic shutdown'
   )
-  .default('name', RUNNER_NAME)
-  .describe('name', 'Name displayed in the repository once registered')
-
+  .default('name')
+  .describe('name', 'Name displayed in the repository once registered cml-{ID}')
+  .coerce('name', (val) => val || RUNNER_NAME)
+  .boolean('retry')
+  .default('retry', RUNNER_RETRY)
+  .describe(
+    'retry',
+    'Automatically retry jobs terminated due to runner disposal or timeout (72 hours on Github)'
```

Comment:
```suggestion
  .boolean('retry-timeout')
  .default('retry-timeout', RUNNER_RETRY)
  .describe(
    'retry-timeout',
    'Automatically terminate and retry after the specified number of minutes (ignored unless on Github)'
```

Reply:
But this approach makes the retrial by default. Do we want this or we want to enforce the users to be asking for it?
@casperdcl @0x2b3bfa0

Reply:
> ignored unless on Github

seems fine to me - basically the default behaviour effectively removes GitHub's timeout

Reply:
Not only that, the concept of retry goes beyond timeout, actually is to cover failure. Here sounds that you can change the timeout time? 

Retry tells me that is going to retry after failure and will also cover the 72 hours timeout in GH. Actually users should dont event know that fact.

I think is better retry for the mentioned reasons. the opposite sounds that you can change the timout time which is fixed or even worse that you might wait a timeout t retry?

Reply:
@casperdcl I do no believe retry-timeout is right because of the reasons already mentioned:
You can not change the timeout. It's always 72 hours, the other variation 3 hours neves hits this since its a runner that its not managed by us. Unless the intention is to force the retry after a few hours and I don't fully understand why that should be the case.

Reply:
ah but users may want to run their own 20min "cleanup" job

Reply:
You mean they want to force the retry after the time they establish? I can not see the benefit of why they would like to have that.... @casperdcl can you see a useful usecase?

Reply:
I would understand the feature max-time-execution but I dont see retry-again-after-30-min? I have to be lossing something. Sorry!

Reply:
```
run: |
  cml-runner --retry-timeout 70h ...

  one_hour_long_cleanup_job

  echo "done"
```

Reply:
API options:

1. (current) `cml-runner --retry`  means kill & restart after 72h only on GH, otherwise never
2. (proposed) `cml-runner --retry-timeout={auto,0,71h,72h,...}` default "auto" means 72h on GH, otherwise 0 (never)

I prefer (2) because in future (or maybe even now) there may be reasons for a user to customise the timeout (e.g. GH: launch a long-running (not spot) cloud instance with code that isn't using DVC cache. User may need some time to do their own caching.)

Basically if supporting (2) is on a long-term roadmap then we should design the CLI API now to be forward-compatible. The proposed naming is also consistent with `--idle-timeout`.

@iterative/cml thoughts?

Reply:
> (proposed) cml-runner --retry-timeout={auto,0,71h,72h,...}

OK now Im lost again 😁 It understood that the timeout was the graceful timeout that we were giving the user BEFORE the 72 hours happened @casperdcl 
So it would be ```--retry-timeout 15m ``` telling that the user would have 15 min before the machine was going to be destroyed because of the 72h limit

Reply:
> default "auto" means 72h on GH, otherwise 0 (never)

the 72 hours clock retry only works in GH, not in any other vendor and time can not be changed

Reply:
Providing an universal `--retry-timeout` (actually `--job-timeout` to be precise) would make sense if we want to ignore GitLab configurable timeouts and offer a consistent CLI API between the different CI/CD vendors.

Reply:
The question is: should users care at all about what we retry? What would the use case be for the proposed `--retry-timeout` option?

Reply:
Im going with retry-unfinished-jobs. Looks clearer the intention and covers every scenario which is restart the workflow if the runner dies during the execution